core/primitives/src/sr25519.rs

```diff
@@ -190,6 +229,27 @@ impl Pair {
 		Pair(kp)
 	}
 
+	/// Make a new key pair from a seed phrase.
+	/// This is generated using schnorrkel's Mini-Secret-Keys.
+	/// A MiniSecretKey is literally what Ed25519 calls a SecretKey, which is just 32 random bytes.
+	pub fn from_entropy(entropy: &[u8], password: Option<&str>) -> Pair {
+		let key = mini_secret_from_entropy(entropy, password.unwrap_or(""))
+			.expect("32 bytes can always build a key; qed");
+		Pair(key.expand_to_keypair::<Sha512>())
+	}
+
+	/// Returns the KeyPair from the English BIP39 seed `phrase`, or `None` if it's invalid.
+	pub fn from_phrase(phrase: &str, password: Option<&str>) -> Option<Pair> {
+		Mnemonic::from_phrase(phrase, Language::English)
+			.ok()
+			.map(|m| Self::from_entropy(m.entropy(), password))
+	}
+
+	/// Recursively derive a child key with some given path.
+	pub fn derive(&self, _path: &[DeriveJunction]) -> Pair {
```

Comment:
I would like either the `DeriveJunction` enum or this function to explain what users can expect from certain inputs.

Reply:
Does it have to be recursive? could be iterative

Reply:
Is this meant to be a "hard" derivation?  A priori, I'd think derivations should only occur in client code, not substrate, but hard derivations remain safe regardless.  If "soft" derivations are used in substrate itself, then we should be careful about what we do with Ed25519 in those places.

Reply:
this is the sr25519 module. ed25519 doesn't have any derivations. there may be instances where substrate needs to derive keys so i'd like to include it in the overall crypto interface.